### PR TITLE
add Transit VPC configuration

### PR DESCRIPTION
**Pull request by afeld**
https://trello.com/c/97xa09nl/530-net-as-the-devsecops-team-i-want-the-transit-vpc-configuration-as-code

This brings the Transit VPC configuration that's currently in the NetOps AWS account under management by Terraform.

While the NetOps account isn't part of the SAIC org, I added a [cross-account access](https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html) to access it in the same way, after [clearing it](https://gsait.slack.com/archives/CAGGRBX42/p1527003197000086) with @ManojChalise, Nadeem, and Zaki first.

## Post-review TODOs

* [ ] ~~Set the new Terraform variables in CircleCI. (Build will fail until this happens.)~~
* [x] Get build green

### CODE REVIEW

File: terraform/master/transit_vpc.tf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            },
+            "Action": [
+                "s3:GetObject",
+                "s3:PutObject",
+                "s3:PutObjectAcl"
+            ],
+            "Resource": "arn:aws:s3:::${aws_cloudformation_stack.transit_vpc.outputs.ConfigS3Bucket}/${aws_cloudformation_stack.transit_vpc.outputs.BucketPrefix}*"
+        }
+    ]
+}
```
Comment (afeld) [inline terraform/master/transit_vpc.tf]: It's outside the scope of this pull request, but realizing that this bucket policy allows any of the spoke accounts to view+change the VPN configuration of any of the other accounts, which seems...bad.

Comment (ManojChalise) [reply terraform/master/transit_vpc.tf]: Agree. Actually the only access needed is for role assumed by spoke lambda. May be we can adjust that later on.

**Comment by kishorebabu12** [comment]
@afeld I did not see the addition of the new sub-account ARN info to KMS key ID (either as a standard set key or can be used as a variable) for Transit VPC hub account. Is this portion intentionally kept out of this build?

**Comment by Vermyndax** [comment]
@kishorebabu12 KMS Key ARNs should be considered sensitive "enough" to be parameterized or created by Terraform as a conditional. They should not be specified directly in hard code.

**Comment by kishorebabu12** [comment]
@Vermyndax You are right. KMS key ARN's are sensitive to be hard coded here. But, this requirement is to add sub-account ARN info to an existing KMS key ID. The goal is to have both these as variable's in our code.

**Comment by Vermyndax** [comment]
@kishorebabu12 that's actually part of what I'm wrestling with in IAM roles. I need to pull in the account ID to dynamically construct the ARN as part of the assume_role_policy.

**Comment by afeld** [comment]
Oooh, good catch on the KMS thing - totally missed that! Working on it, though not seeing a _super_ obvious way to do it.

**Comment by Vermyndax** [comment]
@afeld I dealt with this in other projects (devsecops-ebs-backups) by setting a conditional. If an ARN is given as a variable, use that... if not, create a new KMS resource and use that one instead.

**Comment by afeld** [comment]
Added as a manual step for now, but can revisit to try and automate later.

File: SETUP.md [hunk 2]
```diff
@@ -27,4 +29,7 @@ Having been done once for the current master account, the following steps should
     aws ssm put-parameter --type String --name master-budget --value <budget>
     ```
 
+1. Run a build on the `master` branch of CircleCI.
+1. Connect the Cisco CSR to the GSA network.
```
Comment (afeld) [inline SETUP.md]: Is there a doc or something I can link to?

Comment (ManojChalise) [reply SETUP.md]: @afeld 
Connect the Cisco CSR to the GSA network.
    Step 1:  Coordinate with network operation team to connect Verizon SCI ( Aws virtual interface) to VGW. NetOps will need AWS account number and VGW Id to create ticket to verision to create SCI connection 
Step 2 : Verizon will create sci and VNI appear attached to VGW in transit vpc account
Step 3: Follow detached VGW connection steps documented here. https://docs.aws.amazon.com/solutions/latest/cisco-based-transit-vpc/appendix-d.html

Comment (afeld) [reply SETUP.md]: Added!

**Comment by afeld** [comment]
Build passing! This is good to go, I think.

**Pull request merged**